### PR TITLE
fix: Get QuickPick options from fetched matadata

### PR DESCRIPTION
**Pull request by CsCherrYY**
In spring initializr steps, will get QuickPick options from fetched matadata, instead of hard coding the options. see #170.

### CODE REVIEW

File: src/handler/utils.ts [hunk 1]
```diff
@@ -61,6 +63,8 @@ export async function createPickBox(pickMetadata: IPickMetadata): Promise<boolea
                     pickMetadata.metadata.javaVersion = pickBox.selectedItems[0].label;
                 } else if (pickMetadata.pickStep instanceof SpecifyPackagingStep) {
                     pickMetadata.metadata.packaging = pickBox.selectedItems[0].label && pickBox.selectedItems[0].label.toLowerCase();
+                } else if (pickMetadata.pickStep instanceof SpecifyBootVersionStep) {
+                    pickMetadata.metadata.bootVersion = pickBox.selectedItems[0] && pickBox.selectedItems[0].value && pickBox.selectedItems[0].value.id;
```
Comment (Eskibear) [inline src/handler/utils.ts:67]: can be shortened as `pickBox.selectedItems?.[0]?.value?.id`

File: src/model/ServiceManager.ts [hunk 2]
```diff
@@ -15,15 +16,36 @@ const METADATA_HEADERS = { Accept: "application/vnd.initializr.v2.2+json,applica
 class ServiceManager {
     private metadataMap: Map<string, Metadata> = new Map();
 
-    public async getBootVersions(serviceUrl: string): Promise<any[]> {
+    public async getItems<T extends Identifiable>(serviceUrl: string, type: MatadataType): Promise<Array<IHandlerItem<T>>> {
         const metadata = await this.ensureMetadata(serviceUrl);
         if (!metadata) {
             throw new Error("Failed to fetch metadata.");
         }
+        let defaultLable: string;
```
Comment (Eskibear) [inline src/model/ServiceManager.ts:24]: typo

**Review by Eskibear — CHANGES_REQUESTED** [review]
'chore' is not proper for this PR. please change it.

**Comment by CsCherrYY** [comment]
> 'chore' is not proper for this PR. please change it.

How about `feat`?

**Comment by Eskibear** [comment]
rather than a new "feature", it's a "fix" for bug https://github.com/microsoft/vscode-java-pack/issues/547

**Review by Eskibear — APPROVED** [review]

**Pull request merged**